python/lsst/resources/http.py

```diff
@@ -313,13 +313,35 @@ def _as_local(self) -> Tuple[str, bool]:
         temporary : `bool`
             Always returns `True`. This is always a temporary file.
         """
-        log.debug("Downloading remote resource as local file: %s", self.geturl())
+        log.debug(f"Downloading {self.geturl()} as local file")
```

Comment:
We are not supposed to use f-strings in any log messages since it is deemed to be more efficient to defer the string interpolation until we know the log message will be issued. I think we can probably delete this log message completely since the `time_this` code below is already going to issue a log message.

Reply:
> My main comment is I don't know why we can't work out the block size for the temp file we did get rather than guess in advance where the temp file will end up.

I think it is useful to have the same size of buffer for both reading from the socket and writing to the temporary file. The size of the temporary file buffer is passed to at the creation of the file (ie. the call to `tempfile.NamedTemporaryFile(...)`), this is why it is computed beforehand. The same size is used to read data from the network.

> I'm intrigued that it's more important to get the local file system block size correct than any kind of network transfer chunk size.

I think both are important to reduce system calls: if I read from the network chunks of 4 KB and write to the temporary file chunks of 1 MB, I will need much more `read` system calls than `write` system calls.

Reply:
> We are not supposed to use f-strings in any log messages...

Understood. I didn't know that.